src/commands/publish/mod.rs

```diff
@@ -105,47 +121,35 @@ fn upload_script(
     Ok(())
 }
 
-fn deploy(user: &GlobalUser, target: &Target) -> Result<(), failure::Error> {
-    let routes = target.routes()?;
-    log::info!("routes: {:#?}", &routes);
+fn deploy(user: &GlobalUser, deploy_target: &DeployTarget) -> Result<(), failure::Error> {
+    match deploy_target {
+        DeployTarget::Zoneless(zoneless_config) => {
+            // this is a zoneless deploy
+            log::info!("publishing to workers.dev subdomain");
+            let deploy_address = publish_zoneless(user, zoneless_config)?;
 
-    if routes.is_empty() {
-        // this is a zoneless deploy
-        log::info!("publishing to subdomain");
-        let deploy_address = publish_to_subdomain(target, user)?;
-
-        message::success(&format!(
-            "Successfully published your script to {}",
-            deploy_address
-        ));
+            message::success(&format!(
+                "Successfully published your script to {}",
+                deploy_address
+            ));
 
-        Ok(())
-    } else {
-        // this is a zoned deploy
-        log::info!("publishing to zone");
-        // Most users of Workers Sites will be confused by our route patterns; this warning
-        // informs them of the best configuration for their project, but does not block them
-        // from deploying to a non-wildcard route.
-        if target.site.is_some() && routes.len() == 1 {
-            let route_pattern = &routes[0].pattern;
-            if !route_pattern.ends_with("*") {
-                message::warn(&format!("The route in your wrangler.toml should have a trailing * to apply the Worker on every path, otherwise your site will not behave as expected.\nroute = {}*", route_pattern));
-            }
+            Ok(())
         }
+        DeployTarget::Zoned(zoned_config) => {
+            // this is a zoned deploy
+            log::info!("publishing to zone {}", zoned_config.zone_id);
+
+            let published_routes = publish_routes(&user, zoned_config)?;
 
-        if let Some(zone_id) = &target.zone_id {
-            let published_routes = publish_routes(&user, routes, zone_id)?;
+            let display_results: Vec<String> =
+                published_routes.iter().map(|r| format!("{}", r)).collect();
 
             message::success(&format!(
                 "Deployed to the following routes:\n{}",
-                serde_json::to_string(&published_routes)?
+                display_results.join("\n")
```

Comment:
it would be helpful if in the PR description you included the output of what the command looks like :) - i'm going to check this branch out and do it myself but it would be helpful for other contributors/lurkers